LEAF_Request_Portal/sources/Email.php

```diff
@@ -463,20 +465,313 @@ function getTemplateIDByLabel(string $emailTemplateLabel): int
+    {
+        $empGrpIDs = '';
+        foreach($this->emailFieldsInUse as $indID => $entry) {
+            $isAllowedFormat = $entry['format'] === 'orgchart_employee' || $entry['format'] === 'orgchart_group';
+            if(is_numeric($indID) && $isAllowedFormat) {
+                $empGrpIDs .= $empGrpIDs === '' ? $indID : ', ' . $indID;
+            }
+        }
+
+        if ($empGrpIDs !== '') {
+            $vars = array(
+                ':recordID' => $recordID
+            );
+            $strSQL = "SELECT `indicators`.`format`, `data`.`indicatorID`, `data`.`data` FROM `data`
+                JOIN `indicators` USING (`indicatorID`)
+                WHERE `recordID` = :recordID
+                AND `indicators`.`is_sensitive`=0
+                AND `indicators`.`indicatorID` IN ({$empGrpIDs})";
+
+            $empGrpData = $this->portal_db->prepared_query($strSQL, $vars) ?? [];
+            $plogin = new Login($this->nexus_db, $this->portal_db);
+
+            $emailFields = array();
+            foreach($empGrpData as $rec) {
+                $indID = $rec['indicatorID'];
+                $dataID = $rec['data'];
+                $f = $rec['format'];
+                if ($f === 'orgchart_group') {
+                    $group = new \Portal\Group($this->portal_db, $plogin);
+                    $groupMembers = $group->getMembers($dataID)['data'] ?? [];
+                    $userEmails = array_column($groupMembers, 'email') ?? [];
+                    $emailFields[$indID] = implode("\r\n", $userEmails);
+                }
+                if ($f === 'orgchart_employee') {
+                    $employee = new \Orgchart\Employee($this->nexus_db, $plogin);
+                    $employeeData = $employee->lookupEmpUID($dataID)[0];
+                    $emailFields[$indID] = $employeeData["email"];
+                }
+            }
+            $this->addSmartyVariables(array("field" => $emailFields), true);
+        }
+        $this->emailFieldsInUse = array();
+    }
+
+    /**
+     * format non sensitive data for use in email subject or body based on question type
+     */
+    private function addFormattedData(array $fData, int $recordID): void
+    {
+        foreach($this->fieldsInUse as $k => $entry) {
+            if ((int)$entry['is_sensitive'] === 0 && isset($fData['id' . $k])) {
+                $data = $fData['id' . $k];
+                $f = trim($entry['format']);
+                switch($f) {
+                    case "grid": //process to table
+                        if(isset($fData['id' . $k . '_gridInput'])) {
+                            $data = $fData['id' . $k . '_gridInput'];
+                            $cells = $data['cells'];
+                            $headers = $data['names'];
+
+                            $grid = "<table style=\"border-collapse:collapse;margin:2px;\"><thead><tr>";
+                            foreach($headers as $header) {
+                                if ($header !== " ") {
```

Comment:
These types of comparisons always make me cringe.  I'm thinking it would be better to move the "trim()" function into the foreach, then you can do an "isempty()" check on $header.